iothub_client/samples/pnp/pnp_temperature_controller/pnp_thermostat_component.c

```diff
@@ -6,47 +6,66 @@
 #include <stdlib.h>
 #include <string.h>
 #include <time.h>
+#include <limits.h>
 
 // PnP routines
-#include "pnp_protocol.h"
 #include "pnp_thermostat_component.h"
+#include "iothub_client_properties.h"
 
 // Core IoT SDK utilities
 #include "azure_c_shared_utility/xlogging.h"
 
-// The default temperature to use before any is set
-#define DEFAULT_TEMPERATURE_VALUE 22
+#include "pnp_status_values.h"
 
 // Size of buffer to store ISO 8601 time.
 #define TIME_BUFFER_SIZE 128
 
-// Name of command this component supports to retrieve a report about the component.
-static const char g_getMaxMinReport[] = "getMaxMinReport";
+// Size of buffer to store current temperature telemetry.
+#define CURRENT_TEMPERATURE_BUFFER_SIZE  32
 
-// Names of properties for desired/reporting
+// Size of buffer to store the maximum temp since reboot property.
+#define MAX_TEMPERATURE_SINCE_REBOOT_BUFFER_SIZE 32
+
+// Maximum component size.  The spec (https://github.com/Azure/opendigitaltwins-dtdl/blob/master/DTDL/v2/dtdlv2.md#component)
+// defines the maximum component length as 64 to which we need to reserve the null-terminator.
+#define MAX_COMPONENT_NAME_LENGTH (64 + 1)
+
+// Names of properties for desired/reporting.
 static const char g_targetTemperaturePropertyName[] = "targetTemperature";
 static const char g_maxTempSinceLastRebootPropertyName[] = "maxTempSinceLastReboot";
 
+// Name of command this component supports to retrieve a report about the component.
+static const char g_getMaxMinReport[] = "getMaxMinReport";
```

Comment:
Just as a minor comment, but there are other variables below that are named with suffix "BodyFormat",  "ResponseFormat", etc. That's good, it helps make the code clearer.
Suggestion for variables like this: Add a suffix "CmdName". E.g.: `g_getMaxMinReportCmdName`.

Reply:
Yup, like this too.  Fixing.